### PR TITLE
orm(engine): Implement orm log

### LINKED ISSUE #6465: orm(engine): support self-defined logger for orm
### Is your feature request related to a problem?

No

### Describe the feature you'd like

support self-defined logger for orm

### Describe alternatives you've considered

support self-defined logger for orm

### Teachability, Documentation, Adoption, Migration Strategy

_No response_

### PR DESCRIPTION
**Pull request by maxshuang**
### What problem does this PR solve?

Issue Number: close #6465 

### What is changed and how it works?
Implement logger.Interface to self-defined the orm logger

#### Tests 

 - Unit test

### Release note 

```release-note
 `None`.
```

### CODE REVIEW

File: engine/pkg/orm/log_test.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+	logger.Error(context.TODO(), "%s test", "error")
+	require.Regexp(t, regexp.QuoteMeta("error test"), buffer.Stripped())
+	buffer.Reset()
+
+	fc := func() (sql string, rowsAffected int64) { return "sql test", 10 }
+	logger.Trace(context.TODO(), time.Now(), fc, nil)
+	// expect no log here
+	logger.Trace(context.TODO(), time.Now().Add(-10*time.Second), fc, errors.New("error test"))
+	require.Regexp(t, regexp.QuoteMeta("[sql=\"sql test\"] [affected-rows=10] [error=\"error test\"]"), buffer.Stripped())
```
Comment (amyangfei) [inline engine/pkg/orm/log_test.go:58]: Can test it contains `slow log`

Comment (maxshuang) [reply engine/pkg/orm/log_test.go]: Fix it.

**Review by amyangfei — APPROVED** [review]

File: engine/pkg/orm/log.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+func (l *ormLogger) Warn(ctx context.Context, format string, args ...interface{}) {
+	l.logger.Warn(fmt.Sprintf(format, args...))
+}
+
+func (l *ormLogger) Error(ctx context.Context, format string, args ...interface{}) {
+	l.logger.Error(fmt.Sprintf(format, args...))
+}
+
+func (l *ormLogger) Trace(ctx context.Context, begin time.Time, fc func() (sql string, rowsAffected int64), err error) {
```
Comment (CharlesCheung96) [inline engine/pkg/orm/log.go:73]: `fc` should be more semantic.

File: engine/pkg/orm/util.go [hunk 3]
```diff
@@ -23,3 +34,20 @@ func IsNotFoundError(err error) bool {
...
 }
+
+// NewGormDB news a gorm.DB
+func NewGormDB(sqlDB *sql.DB) (*gorm.DB, error) {
+	db, err := gorm.Open(mysql.New(mysql.Config{
+		Conn:                      sqlDB,
+		SkipInitializeWithVersion: false,
+	}), &gorm.Config{
+		SkipDefaultTransaction: true,
+		Logger:                 NewOrmLogger(logutil.WithComponent("gorm"), WithSlowThreshold(200*time.Millisecond)),
```
Comment (CharlesCheung96) [inline engine/pkg/orm/util.go:45]: Better to use `defaultSlowThreshold`

File: engine/pkg/orm/log_test.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+func TestNewOrmLogger(t *testing.T) {
+	t.Parallel()
+
+	var buffer zaptest.Buffer
+	lg, _, err := log.InitLoggerWithWriteSyncer(&log.Config{Level: "warn"}, &buffer, nil)
+	require.NoError(t, err)
+
+	logger := NewOrmLogger(lg, WithSlowThreshold(3*time.Second))
+	logger.Info(context.TODO(), "%s test", "info")
+	// expect no log here
```
Comment (CharlesCheung96) [inline engine/pkg/orm/log_test.go:45]:
```suggestion
	// expect no log here
	require.Equal(t, 0, len(buffer.Lines()))
```

File: engine/pkg/orm/log.go (hunk 2 above)
Comment (maxshuang) [reply engine/pkg/orm/log.go]: The signature of the function is semantic.
It means this function will return the `sql` and `affected-rows`.

File: engine/pkg/orm/util.go (hunk 3 above)
Comment (maxshuang) [reply engine/pkg/orm/util.go]: Fix it.

File: engine/pkg/orm/log_test.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+	require.Regexp(t, regexp.QuoteMeta("warn test"), buffer.Stripped())
+	buffer.Reset()
+
+	logger.Error(context.TODO(), "%s test", "error")
+	require.Regexp(t, regexp.QuoteMeta("error test"), buffer.Stripped())
+	buffer.Reset()
+
+	fc := func() (sql string, rowsAffected int64) { return "sql test", 10 }
+	logger.Trace(context.TODO(), time.Now(), fc, nil)
+	// expect no log here
```
Comment (CharlesCheung96) [inline engine/pkg/orm/log_test.go:56]:
```suggestion
	// expect no log here
	require.Equal(t, 0, len(buffer.Lines()))
```

File: engine/pkg/orm/log_test.go (hunk 4 above)
Comment (maxshuang) [reply engine/pkg/orm/log_test.go]: Fix it.

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**